src/legacy/core_plugins/visualizations/public/embeddable/visualize_embeddable.ts

```diff
@@ -301,18 +290,13 @@ export class VisualizeEmbeddable extends Embeddable<VisualizeInput, VisualizeOut
 
     this.subscriptions.push(
       this.handler.events$.subscribe(async event => {
-        if (this.actions[event.name]) {
-          event.data.aggConfigs = getTableAggs(this.vis);
-          const filters: esFilters.Filter[] = this.actions[event.name](event.data) || [];
-          const mappedFilters = mapAndFlattenFilters(filters);
-          const timeFieldName = this.vis.indexPattern.timeFieldName;
-
-          npStart.plugins.uiActions.executeTriggerActions(APPLY_FILTER_TRIGGER, {
-            embeddable: this,
-            filters: mappedFilters,
-            timeFieldName,
-          });
-        }
+        const eventName = event.name === 'brush' ? SELECT_RANGE_TRIGGER : APPLY_FILTER_TRIGGER;
```

Comment:
I am not sure we should separate these out instead of having a single one.

If we think the user will want something different to happen based on which one is emitted, we should separate them, but if in the users head, the same thing should happen, we should leave them together.

Right now, the user see's the same exact thing happen, the only difference here is in the code, to construct a different filter object, but in either case, a filter gets added to the view.  But now, if the user wants to add a drilldown, we have two triggers, and we don't really know which one will be emitted from any given visualization.  We could try to expose it but that will be difficult with lens when you don't really have the information from the renderer. 

One other option is to leave these triggers separately, name them something like you suggested, SELECT_RANGE and CLICKED_DATA_POINT, and actually have each of those actions emit _another_ trigger, maybe named ES_FILTER_SELECTED_TRIGGER.  Then drilldowns actions can attach to that trigger name, and it won't matter whether it was a brush or a click, it only matters that the end result was that the user selected some filters, and any relevant drilldowns will apply.

if we wanted to do that, and eventually allow users to attach the raw URL builder drilldown where we will expose the raw data, not the es filter data, we'd need to figure out a way to batch or collect all triggers being emitted in a single event cycle.  Otherwise, the user could attach an action on both CLICKED_DATA_POINT, and FILTER_SELECTED, and since they were separately emitted as part of the same event cycle, you'd see two context menu's pop up and we don't want that.

I think probably to avoid complicating drilldown feature, for phase 1, we should have at least one common trigger we can attach the drilldowns to, and not worry about having to figure out whether to attach the drilldown to the select range or data clicked trigger.  Then once we expose the ability to attach actions to the triggers that have the raw data, figure out the batching then. 

Thoughts?

Reply:
i really like your idea about this actions emitting yet another trigger `ES_FILTER_SELECTED_TRIGGER`

Reply:
i will leave creating another trigger `ES_FILTER_SELECTED_TRIGGER) out of this PR, but i agree that could be a possible solution to simplify the drilldown (or any user defined action) configuration.